domains/transfer/TransferAgreementAutomation/Worker/Program.cs

```diff
@@ -0,0 +1,125 @@
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using ProjectOrigin.WalletSystem.V1;
+using Serilog;
+using Serilog.Enrichers.Span;
+using Serilog.Formatting.Json;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using TransferAgreementAutomation.Worker;
+using TransferAgreementAutomation.Worker.Metrics;
+using TransferAgreementAutomation.Worker.Options;
+using TransferAgreementAutomation.Worker.Service;
+using TransferAgreementAutomation.Worker.Swagger;
+
+
+var builder = WebApplication.CreateBuilder(args);
+var loggerConfiguration = new LoggerConfiguration()
+    .Filter.ByExcluding("RequestPath like '/health%'")
+    .Filter.ByExcluding("RequestPath like '/metrics%'")
+    .Enrich.WithSpan();
+
+loggerConfiguration = builder.Environment.IsDevelopment()
+    ? loggerConfiguration.WriteTo.Console()
+    : loggerConfiguration.WriteTo.Console(new JsonFormatter());
+
+builder.Logging.ClearProviders();
+builder.Logging.AddSerilog(loggerConfiguration.CreateLogger());
+
+builder.Services.AddOptions<OtlpOptions>().BindConfiguration(OtlpOptions.Prefix).ValidateDataAnnotations()
+    .ValidateOnStart();
+builder.Services.AddOptions<TransferApiOptions>().BindConfiguration(TransferApiOptions.TransferApi)
+    .ValidateDataAnnotations().ValidateOnStart();
+builder.Services.AddOptions<ProjectOriginOptions>().BindConfiguration(ProjectOriginOptions.ProjectOrigin)
+    .ValidateDataAnnotations().ValidateOnStart();
+
+var otlpConfiguration = builder.Configuration.GetSection(OtlpOptions.Prefix);
+var otlpOptions = otlpConfiguration.Get<OtlpOptions>()!;
+
+builder.Services.AddOpenTelemetry()
+    .WithMetrics(provider =>
+        provider
+            .SetResourceBuilder(ResourceBuilder.CreateDefault()
+                .AddService(TransferAgreementAutomationMetrics.MetricName))
+            .AddMeter(TransferAgreementAutomationMetrics.MetricName)
+            .AddHttpClientInstrumentation()
+            .AddAspNetCoreInstrumentation()
+            .AddRuntimeInstrumentation()
+            .AddProcessInstrumentation()
+            .AddOtlpExporter(o => o.Endpoint = otlpOptions.ReceiverEndpoint));
+
+builder.Services.AddControllers().AddJsonOptions(options =>
+    options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter()));
+
+builder.Services.AddAuthorization();
+builder.Services.AddHealthChecks();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddHttpContextAccessor();
+builder.Services.AddLogging();
+
+builder.Services.AddHostedService<TransferAgreementsAutomationWorker>();
+builder.Services.AddSingleton<AutomationCache>();
+builder.Services.AddSingleton<ITransferAgreementAutomationMetrics, TransferAgreementAutomationMetrics>();
+builder.Services.AddSingleton<IProjectOriginWalletService, ProjectOriginWalletService>();
+
+builder.Services.AddHttpClient<TransferAgreementsAutomationWorker>();
```

Comment:
Is this TransferAgreementsAutomationWorker a HttpClient?

Reply:
It is to inject the httpfactory to the worker. We still create a hosted service above